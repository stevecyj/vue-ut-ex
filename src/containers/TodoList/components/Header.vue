<script setup>
import { ref, defineEmits, computed, nextTick } from "vue";
import { useStore } from "vuex";

const store = useStore();
defineProps({});

// const inputValue = ref("");
const inputValue = computed(() => {
  return store.getters.inputValue;
});

const emit = defineEmits(["add"]);

const changeInputValue = (value) => {
  store.commit("changeInputValue", value);
};

const addTodoItem = () => {
  // emit add event with inputValue
  if (!inputValue || inputValue == "") return;
  emit("add", inputValue);
  changeInputValue("");
};
</script>

<template>
  <div class="header">
    <div class="header-content">
      TodoList
      <input
        class="header-input"
        type="text"
        data-test="header-input"
        :value="inputValue"
        @input="(e) => changeInputValue(e.target.value)"
        @keyup.enter="addTodoItem"
        placeholder="Please input your todo item !"
      />
    </div>
  </div>
</template>

<style lang="scss" scoped>
.header {
  line-height: 60px;
  background: #333;
  color: #fff;
  font-size: 24px;

  &-content {
    width: 600px;
    margin: 0 auto;
  }
  &-input {
    float: right;
    width: 360px;
    margin-top: 16px;
    line-height: 24px;
    outline: none;
    color: #333;
    text-indent: 10px;
  }
}
</style>
