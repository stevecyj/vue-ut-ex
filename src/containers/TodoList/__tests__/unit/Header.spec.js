// import HelloWorld from "./HelloWorld.vue";
import Header from "@/containers/TodoList/components/Header.vue";
import { shallowMount } from "@vue/test-utils";

it("Header 包含 input 框", () => {
  const wrapper = shallowMount(Header);
  const input = wrapper.find('[data-test="input"]');
  expect(input.exists()).toBe(true);
});

it("Header input init value ''", () => {
  const wrapper = shallowMount(Header);
  const inputValue = wrapper.vm.inputValue;
  console.log("inputValue", inputValue);
  expect(inputValue).toBe("");
});

it("Header input input值發生變化，數據應該跟著變", () => {
  const wrapper = shallowMount(Header);
  const input = wrapper.find('[data-test="input"]');
  input.setValue("123");
  const inputValue = wrapper.vm.inputValue;
  expect(inputValue).toBe("123");
});

it("Header input 框輸入 enter，無內容時無反應", () => {
  const wrapper = shallowMount(Header);
  const input = wrapper.find('[data-test="input"]');
  input.setValue("");
  input.trigger("keyup.enter");
  expect(wrapper.emitted().add).toBeFalsy();
});
