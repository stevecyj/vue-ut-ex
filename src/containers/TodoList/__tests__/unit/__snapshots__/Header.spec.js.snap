// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Header 樣式發生改變時提示 1`] = `
<div class="header">
  <div class="header-content"> TodoList <input class="header-input" type="text" placeholder="Please input your todo item !"></div>
</div>
`;
